### PR TITLE
tests: smtp frames tests

### PR DESCRIPTION
**Pull request by victorjulien**
#1857 rebased

### CODE REVIEW

File: tests/smtp-long-DATA-line-02-frames/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+This test shows how we handle long DATA lines for SMTP.
+
+## PCAP
+
+PCAP comes from ttps://osqa-ask.wireshark.org/questions/33094/extract-an-attachment-email-smtp-cap
+and has been modified to have a really long DATA line (6512 Bytes).
+
+## Related issues
+
+https://redmine.openinfosecfoundation.org/issues/5981
```
Comment (catenacyber) [inline tests/smtp-long-DATA-line-02-frames/README.md:12]: 4905 rather

File: tests/smtp-long-DATA-line-02-frames/test.rules [hunk 2]
```diff
@@ -0,0 +1,6 @@
+alert smtp any any -> any any (frame:smtp.command_line; content:"MAIL|20|FROM:"; startswith; content:"xxxxx.co.uk"; distance:0; content:">|0d 0a|"; endswith; sid:1;)
+alert smtp any any -> any any (frame:smtp.data; content:"Reply-To:"; startswith; content:"Subject"; distance:0; content:"This is a multi-part message in MIME format."; distance:0; sid:2;)
+alert smtp any any -> any any (frame:smtp.response_line; content:"220 smtp001.mail.xxx.xxxxx.com ESMTP"; startswith; sid:3;)
+alert smtp any any -> any any (frame:smtp.stream; content:"|0d 0a|.|0d 0a|"; sid:4;)
```
Comment (catenacyber) [inline tests/smtp-long-DATA-line-02-frames/test.rules:4]: Why do not I see this one in `SMTPFrameTypes` ?

File: tests/smtp-long-DATA-line-03-frames-ips/suricata.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+%YAML 1.1
```
Comment (catenacyber) [inline tests/smtp-long-DATA-line-03-frames-ips/suricata.yaml:1]: This is the same pcap that can be deduplicated, right ?

File: tests/smtp-long-DATA-line-03-frames-ips/test.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- -k none
+- --simulate-ips
+
+checks:
+- filter:
+    count: 1
+    match:
+      alert.signature_id: 1
+- filter:
+    count: 2 # TODO 2 because of multiple smtp.data updates, and we're doing sliding window
```
Comment (catenacyber) [inline tests/smtp-long-DATA-line-03-frames-ips/test.yaml:14]: Should one frame match only once per signature as is already done for transactions ?

File: tests/smtp-long-DATA-line-02-frames/test.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+checks:
+- filter:
+    count: 1
+    match:
+      alert.signature_id: 1
+- filter:
+    count: 1
+    match:
+      alert.signature_id: 2
```
Comment (catenacyber) [inline tests/smtp-long-DATA-line-02-frames/test.yaml:15]: We should also test app-layer metadata logging

File: tests/smtp-long-DATA-line-02-frames/test.rules (hunk 2 above)
Comment (victorjulien) [reply tests/smtp-long-DATA-line-02-frames/test.rules:4]: it's a built in frame that is created for each (tcp) app-layer

File: tests/smtp-long-DATA-line-02-frames/README.md (hunk 1 above)
Comment (victorjulien) [reply tests/smtp-long-DATA-line-02-frames/README.md:12]: adding

File: tests/smtp-long-DATA-line-02-frames/README.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# Test Description
+
+This test shows how we handle long DATA lines for SMTP.
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply tests/smtp-long-DATA-line-02-frames/README.md:3]: ya

File: tests/smtp-long-DATA-line-03-frames-ips/suricata.yaml (hunk 3 above)
Comment (victorjulien) [reply tests/smtp-long-DATA-line-03-frames-ips/suricata.yaml:1]: done now

File: tests/smtp-long-DATA-line-03-frames-ips/test.yaml (hunk 4 above)
Comment (victorjulien) [reply tests/smtp-long-DATA-line-03-frames-ips/test.yaml:14]: maybe, but that is out of scope for this work. It's a general streaming buffer issue.

File: tests/smtp-long-DATA-line-02-frames/test.yaml (hunk 5 above)
Comment (victorjulien) [reply tests/smtp-long-DATA-line-02-frames/test.yaml:15]: adding

File: tests/smtp-long-DATA-line-02-frames/test.rules (hunk 2 above)
Comment (catenacyber) [reply tests/smtp-long-DATA-line-02-frames/test.rules:4]: Is it documented ? (I cannot find it)
Where does it start ? Where does it end ?

File: tests/smtp-long-DATA-line-03-frames-ips/test.yaml (hunk 4 above)
Comment (catenacyber) [reply tests/smtp-long-DATA-line-03-frames-ips/test.yaml:14]: Ok

Or if a frame matches and there is a tx associated to it, we can add the sid in the engine state there...

File: tests/smtp-long-DATA-line-02-frames/test.rules (hunk 2 above)
Comment (victorjulien) [reply tests/smtp-long-DATA-line-02-frames/test.rules:4]: Starts at start of stream, ends at end of stream.

It's essentially a frame that is the entire stream, but exposed as <appproto>.stream. Behaves like:
```
alert <app proto> ... flow:only_stream; content:"abc"; 
```

Comment (catenacyber) [reply tests/smtp-long-DATA-line-02-frames/test.rules:4]: Thanks for the explanation

**Comment by victorjulien** [comment]
replaced by #1881

**Pull request closed without merging**